pandas/core/dtypes/dtypes.py

```diff
@@ -13,6 +24,15 @@
 from pandas.core.dtypes.generic import ABCCategoricalIndex, ABCDateOffset, ABCIndexClass
 from pandas.core.dtypes.inference import is_bool, is_list_like
 
+if TYPE_CHECKING:
+    import pyarrow  # noqa: F401
```

Comment:
This means that running mypy (eg in a pre-commit hook) requires pyarrow to be installed? (which is not a required dependency?) 
Are we fine with that? (do we already do that for other deps?)

Reply:
> This means that running mypy (eg in a pre-commit hook) requires pyarrow to be installed?

no. mypy is a static checker.

The behavior if pyarrow is not installed depends on the strictness of the type checking.(ignore_missing_imports config option)

in general,"mypy will assume the type of that module is Any, the dynamic type. This means attempting to access any attribute of the module will automatically succeed" 
 https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports